backends/audiocd/macosx/macosx-audiocd.cpp

```diff
@@ -195,16 +197,19 @@ enum {
 
 MacOSXAudioCDManager::DriveList MacOSXAudioCDManager::detectAllDrives() {
 	// Fetch the lists of drives
-	struct statfs driveStats[kMaxDriveCount];
-	int foundDrives = getfsstat(driveStats, sizeof(driveStats), MNT_WAIT);
-	if (foundDrives <= 0)
+	struct statfs *driveStats = (struct statfs *)malloc(sizeof(struct statfs) * kMaxDriveCount);
+	int foundDrives = getfsstat(driveStats, sizeof(struct statfs) * kMaxDriveCount, MNT_WAIT);
```

Comment:
I don't know if this is worth it, but with allocation on the heap we could actually get rid of `kMaxDriveCount` and allocate only what is needed:
```c++
int nbDrives = getfsstat(nullptr, 0, MNT_WAIT);
struct statfs *driveStats = (struct statfs *)malloc(sizeof(struct statfs) * nbDrives);
nbDrives = getfsstat(driveStats, sizeof(struct statfs) * nbDrives, MNT_WAIT);
```
As documented `getfsstat()` can be called with a null buffer to get the number of drives.
And I am not sure (so I didn't do it above), but it's possible the second call could actually be switched to `MNT_NOWAIT` as we could expect the first call to ensure the information held in the kernel are updated if needed.

Reply:
@criezy: Thanks. I see what you mean. I've pushed a new commit in this PR, trying to do that (although untested with a real audio CD for now; so marking this PR as a draft for now).

What do you think about it? It looks like this Rust project from Google did something similar: https://github.com/google/rrg/blob/fcecd6ad8f550100021855a5b180d1be3039ddec/crates/ospect/src/fs/macos.rs#L168-L210 (although they always use MNT_NOWAIT for some reason)

I just wonder whether we're not taking some risk for a small gain, though. Allocating a big buffer is maybe a bit wasteful, but then we avoid the risks of two subsequent `getfsstat()` apparently having no coherence guarantee.

Reply:
The "big" buffer is not that big (less than 30 kb I think), so indeed I am not sure there is much gain in trying to be smart. I don't think there is much risk either, but if there is no gain we might as well go with the simplest code. One possible benefit of using two calls is that this would work if we somehow have more than 256 drives, but that seems unlikely to happen.

Reply:
OK, well, let's try just using your suggestion above, then.

i.e. try getting rid of `kMaxDriveCount` by making two consecutive `getfsstat()` calls, but keeping this as simple as possible (i.e. without the `MIN()` stuff you reviewed below)